javalib/src/main/scala/java/net/AbstractPlainSocketImpl.scala

```diff
@@ -122,7 +124,9 @@ private[net] abstract class AbstractPlainSocketImpl extends SocketImpl {
     val src = inetAddress.getAddress()
 
     if (inetAddress.isInstanceOf[Inet6Address]) {
-      sa6.sin6_addr = src.asInstanceOf[in.in6_addr]
+      val from = src.asInstanceOf[scala.scalanative.runtime.Array[Byte]].at(0)
+      val dst = sa6.sin6_addr.at1.at(0).asInstanceOf[Ptr[Byte]]
+      memcpy(dst, from, 16.toUInt)
```

Comment:
Would you mind adapting these lines to the Unsafe Arrays Ops and it's new `at` method before the merge? It should make these lines a bit cleaner

Reply:
Using the new `at` method is my plan.  I have been checking the merge stream.  PR #2888 hit the 0.4.5
but has not yet been rolled forward to the 0.5.0 stream.  I do not want to rush the maintainer, who is
busy merging.

I think there are also a couple of other places I can use `at`, when it gets merged into 0.5.0 in the 
fullness of time.

This PR contains a number of changes. I am confident in it, but would like to see it exercised on 0.5.0
for awhile before being backported.   This is especially true hours before a/an 0.4.n release.
Yes, this is an exceedingly conservative approach.   

I know of nothing which precludes backporting, but have never tried this PR on 0.4.n.

Reply:
Oh, sorry, I forget it was targeting the 0.4.x branch. In such a case, I don't want you to block until we would sync 0.4.x and the main branches again. Instead, let's merge it now and apply a fix when changes would be backported.

Reply:
Thank you for the merge.

| Oh, sorry, I forget it was targeting the 0.4.x branch.

It is the other way around.  This is 0.5.0 until it earns its keep.
I do not envy you sorting through the complexity.

I will keep my eyes peeled for when #2888 merges and apply
the fix.  I would much rather that @Arman be blamed for 
spraying memory than I ;-)